pax/simulation.py

```diff
@@ -801,3 +801,29 @@ def truncated_gauss_rvs(my_mean, my_std, left_boundary, right_boundary, n_rvs):
     See http://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.truncnorm.html
     """
     return _truncated_gauss(my_mean, my_std, left_boundary, right_boundary).rvs(n_rvs) * my_std + my_mean
+
+def load_spe_pdf(file):
+    """Extract the spe pdf from a csv file in form of bin, prob. Returns two arrays."""
+    charge = []
+    p = []
+    with open(file, "r") as f:
+        for line in f.readlines():
+            q = float(line.split(",")[0])
+            prob = float(line.split(",")[1])
+            # don't allow negative charge, which can occur from noise subtraction
+            prob = prob if c >= 0 else 0
+            charge.append(q)
+            p.append(prob)
+
+    return np.array(charge), np.array(p)
+
+def generate_gains(bins, pdf, N):
+    """ Take the charge bins, pdf of spe distribution and generate sample of N gains"""
+    cdfs = np.cumsum(pdf)/np.sum(pdf)
+    Interpolator = interp1d(cdfs, bins)
+    random_cdfs = np.random.uniform(0, 1, N)
+    gains = []
+    for cdf in random_cdfs:
+        gains.append( Interpolator(cdf))
+    return gains
+
```

Comment:
__[flake8]__
* [W391] blank line at end of file

<!-- Comment by [SideCI](https://sideci.com) -->